### PR TITLE
Deposit using SDR API

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

One of the goals for H2 is to develop an API for doing deposits.  If we make the GUI use that API it validates that the API is useful and complete and means we only have to support a single path for both GUI or API deposits.  This will allow us to record "who" is doing the deposit, because in the future we can use the users credentials (rather than the application credentials).

### CODE REVIEW

File: app/jobs/deposit_job.rb [hunk 1]
```diff
@@ -3,16 +3,49 @@
...
+
   queue_as :default
 
+  sig { params(work: Work).void }
   def perform(work)
-    result = api_client.objects.register(params: create_model(work))
-    work.druid = result.externalIdentifier
+    job_id = deposit(create_model(work))
+    result = nil
+    until result
```
Comment (mjgiarlo) [inline app/jobs/deposit_job.rb:14]: Not for this PR, but how would you feel about follow-up work that lets sdr-client manage its async interactions on apps' behalf, like dor-services-client does? https://github.com/sul-dlss/dor-services-client#asynchronous-results That will simplify the DepositJob.

File: config/settings.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
-dor_services:
-  url: ~
-  token: ~
+sdr_api:
```
Comment (mjgiarlo) [inline config/settings.yml:1]: And confirming here: we already have this in shared_configs for all three deployed environments.

**Review by mjgiarlo — APPROVED** [review]
LGTM. Can you leave a little more detail in the PR description going forward? Link to an issue, some description of the rationale (for new folks joining the WC), answering the questions in the template, etc. Thanks!

File: app/jobs/deposit_job.rb (hunk 1 above)
Comment (jcoyne) [reply app/jobs/deposit_job.rb:14]: Sounds good to me.

File: config/settings.yml (hunk 2 above)
Comment (jcoyne) [reply config/settings.yml:1]: exactly.

**Pull request merged**